src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/sdo/SdoGeneratorService.java

```diff
@@ -53,9 +53,12 @@ public CaseDocument generate(CaseData caseData, String authorisation) {
                 .anyMatch(s -> s != null && s.toLowerCase().contains("judge"));
         }
 
-        if (SdoHelper.isSmallClaimsTrack(caseData)) {
+        if (SdoHelper.isSmallClaimsTrack(caseData) && SdoHelper.hasSmallAdditionalDirections(caseData, "smallClaimFlightDelay")) {
```

Comment:
I would have only tested featureToggleService.isSdoR2Enabled() and SdoHelper.isSmallClaimsTrack(caseData).  if  featureToggleService.isSdoR2Enabled() is true then use new template otherwise old use template. Because when this feature will go live we will be only using new template "01388". This check " SdoHelper.hasSmallAdditionalDirections(caseData, "smallClaimFlightDelay")" will not be required when we go live and shouldnt be required for selecting new template

Reply:
Updated the code to check only isSdoR2Enabled now.